### ISSUE TITLE
missing t_PE & t_PS seq chars when paste in gnu-screen?

### ISSUE THREAD

**Issue by Shane-XB-Qian** [vim/vim#10632]
### Steps to reproduce

1. run vim with --clean and log in xfce-term (or some else)
here is the chars when paste 'foo' into vim in insert mode : `  4.726571 : raw key input: "[200~foo[201~"`

2. run vim with --clean and log in xfce-term (or some else) with gnu-screen
here is the chars when paste 'bar' into vim in insert mode: `  2.588195 : raw key input: "bar"`

apparently t_PE and t_PS did not appear at gnu-screen way.
// if echo &t_PE and &t_PS there is its value.

### Expected behaviour

not sure which side should response to input t_PE and t_PS chars.
vim? or gnu-screen?
// perhaps not a bug, or more like a question.

### Version of Vim

v8.2.5172

### Environment

linux

**Comment by Shane-XB-Qian** [comment]
yes. and apparently those doc (of bracketed-paste) for Gun-screen is not actually working.
and there was a trick to inoremap the t_PS to trigger `set paste` option, but if there no t_PS sea chars was presented, then looks no chance to implement that (if I was understood cortectly)

--
shane.xb.qian
________________________________

**Comment by lacygoill** [comment]
Did you set `'t_BE'` and `'t_BD'`?
```vim
if &term =~ 'screen'
    let &t_BE = "\<Esc>[?2004h"
    let &t_BD = "\<Esc>[?2004l"
    let &t_PS = "\<Esc>[200~"
    let &t_PE = "\<Esc>[201~"
endif
```
What happens if – in GNU screen – you start Vim with a modified `TERM` so that Vim knows that the terminal is compatible with xterm and supports the feature; e.g.:

    $ TERM=xterm-256color vim --clean

**Comment by Shane-XB-Qian** [comment]
> Did you set `'t_BE'` and `'t_BD'`?

those `let` would not change something (not working).


> start Vim with a modified `TERM`

i think it is not a good idea, specially user may like to judge cur env by `&term =~ 'screen'`
but anyway, it did not change something even set it.

**Comment by lacygoill** [comment]
> those let would not change something (not working).

Yes, now that I've made more tests, I can confirm it doesn't help.

   > i think it is not a good idea

I was not suggesting that as a solution, but as a way to gather more information.

---

I wonder whether GNU screen supports bracketed paste, and if it does since which version, and if it needs some config or just works out of the box.

I found this [somewhat recent message](https://savannah.gnu.org/bugs/?func=detailitem&item_id=62102) on the bug tracker for the project:

   > Bracketed paste mode has been in master since 2013. It should be in V4 too. I have prepared patches that apply cleanly to the current screen-v4 branch.

The first sentence suggests the feature has been supported for a long time; so not sure why patches need to be applied.

**Comment by Shane-XB-Qian** [comment]
> I wonder whether GNU screen supports bracketed paste, and if it does since which version, and if it needs some config or just works out of the box.

good question!
seems it had been committed to branch since `Mar 15 2013`

// there was a *trick* (above comment said) to trigger `set paste` but if no such t_PS & t_PE seq chars presened then no chance to impl it.

**Comment by Shane-XB-Qian** [comment]
BTW: gnu-screen v4.0.9 solved the issue (happened at v4.0.8) #10523
However the master branch of gnu-screen bring the issue back....... what a amazing release manager..............................
// but whatever v4.0.9 made vim cursorline/cursorcol highlight a bit abnormal..
// or master branch made its status highlight (host in terminal (e.g xfce-term)) a bit abnormal..

**Comment by Shane-XB-Qian** [comment]
supposed this was a gnu-screen issue. 

// similar https://github.com/vim/vim/issues/10523 etc
// or actually not sure which really caused it..
